### PR TITLE
UniFi AP and UAP AC Lite

### PR DESCRIPTION
**Pull request by robimarko**
Pull request for adding UAP AC Lite support.
In future I will add remaining Unifi AP-s but we use this one the most.

### CODE REVIEW

**Comment by robimarko** [comment]
Well,after fixing syntax build fails on IEEE80211AC for Radio0.
What should I instead use of AC since device has AC radio.

**Comment by kostko** [comment]
Currently, there is no support for AC. It should be easy to add, but it must be done at several places (protocols, and CGM for the OpenWrt/LEDE platform).

**Comment by robimarko** [comment]
Do you have example so I can try to add it?
We currently use a lot of UAP AC Lites for Otvorena mreza and it would be nice to have support for them without Unifi controller

**Comment by mitar** [comment]
This is really great! I would love to see support for UniFi AC Pro as well. And [UniFi AP Pro](https://wiki.openwrt.org/toh/ubiquiti/unifi_appro) (old model). OpenWrt works great on the old model (see [here how I installed it](https://github.com/cloyne/network/tree/master/unifi)). I have not yet tested UniFi AC Pro.

What is difference between lite and pro?

Also, can you name this "UniFi AP AC Lite"?

**Comment by mitar** [comment]
I also have a setting where I would like to use all of those without a controller. It would be great if they would be supported in nodewatcher.

**Comment by robimarko** [comment]
Well,difference is mostly in antennas.
Pro has 3x3 MIMO for 2.4 and 3x3 MIMO for 5GHz instead of 2x2 for 2.4GHz and 2x2 for 5GHz in Lite.
Also PRO has 2 gigabit ports and USB port and is powered by standard 802.3af or 802.3at POE instead of passive 24V.

It will just require different profile because of the 2 ports.
I can make profiles for UAP Pro, UAP AC LR and UAP AC Pro but I cant test those.

Since UAP AC Pro has AC support like Lite it requires Nodewatcher AC support.
Can you review my Edgerouter X profile since I totally reworked it today as per LEDE?

Also if you can check "uci show network output" on UAP Pro?

**Comment by robimarko** [comment]
I have added first gen UAP profile also.
Same profile can be used for UAP-LR

**Comment by mitar** [comment]
BTW. You can check PEP8 before you commit. :-) You can even use an IDE which does it automatically for you. (Be careful, we do use a bit relaxed PEP8 so use our PEP8 configuration.)

**Comment by robimarko** [comment]
Yeah,I gotta start using that.
It would be great If at least ordinary first gen UAP can be merged so I can test it on UAP and UAP-LR

**Comment by kostko** [comment]
> Do you have example so I can try to add it?

I can add it, just let me know how to configure AC in `/etc/config/wireless` on LEDE/OpenWrt?

File: nodewatcher/modules/devices/ubnt/unifi.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
+from django.utils.translation import ugettext_lazy as _
+
+from nodewatcher.core.generator.cgm import base as cgm_base, protocols as cgm_protocols, devices as cgm_devices
+
+
+class UBNTUapACLite(cgm_devices.DeviceBase):
```
Comment (mitar) [inline nodewatcher/modules/devices/ubnt/unifi.py:6]: I have some issues with capitalization of this class name.

File: nodewatcher/modules/devices/ubnt/unifi.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    """
+    UBNT UAP AC Lite device descriptor.
```
Comment (mitar) [inline nodewatcher/modules/devices/ubnt/unifi.py:8]: Can you sync the name in the comment with name in the class?

**Comment by mitar** [comment]
So if you prepare descriptors for UniFi AP Pro and UniFi AP AC Pro I can test them.

@CdavM, do we have any UniFi AP AC Pro around to test? I have one UniFi AP Pro in Cloyne I can test on.

**Comment by robimarko** [comment]
@kostko Here is output from /etc/config/wireless on UAP AC Lite running LEDE.
[wireless.txt](https://github.com/wlanslovenija/nodewatcher/files/998455/wireless.txt)

Radio0 is 5Ghz

@mitar I cant think how to make capitalization better due to weird name.
Also I will try to make a profile for UAP AC Pro and UAP Pro today or tommorow,but having AC support in Nodewatcher is needed

**Comment by kostko** [comment]
Preliminary 802.11ac protocol support added in #37, please check if it works correctly and then we can merge it into development branch.

**Comment by robimarko** [comment]
From what I have seen it should work.
How can I check it?

**Comment by kostko** [comment]
The best way is to [setup a local nodewatcher instance](http://docs.nodewatcher.net/en/development/development.html) with code from that `feature/80211ac` branch and try your device descriptors with `IEEE80211AC` protocol specified (e.g., try to generate firmware for these new devices and check that the configuration matches).

**Comment by robimarko** [comment]
Well that could be an issue.
Do you have documentation on setting up local Nodewatcher so I can set one up in VM.

**Comment by kostko** [comment]
Yes, I've linked it above :-)

**Comment by robimarko** [comment]
Yes,I was stupid enough not to see hyperlink.
Will try today if I manage to setup Nodewatcher.

Can you check out my pull request for Edgerouter X,yesterday I reworked it so it is per LEDE configuration and should be correct.

**Comment by kostko** [comment]
Ok, I've tested 802.11ac support and it appears to work so I've merged it into master. Please rebase your pull request on `development`.

**Comment by robimarko** [comment]
Should be done

**Comment by kostko** [comment]
You merged instead of rebasing, please see my comment in #35, also on how to do squashing.

**Comment by robimarko** [comment]
Should be done now

**Comment by mitar** [comment]
> I can make profiles for UAP Pro, UAP AC LR and UAP AC Pro but I cant test those.

I can test UniFi AP Pro, UniFi AP LR and UniFi AC Pro, but not UniFi AC LR.

> Also if you can check "uci show network output" on UAP Pro?

```
network.loopback=interface
network.loopback.ifname=lo
network.loopback.proto=static
network.loopback.ipaddr=127.0.0.1
network.loopback.netmask=255.0.0.0
network.globals=globals
network.globals.ula_prefix=fd28:7dda:d196::/48
network.lan=interface
network.lan.ifname=eth0
network.lan.force_link=1
network.lan.type=bridge
network.lan.proto=static
network.lan.ipaddr=10.20.32.40
network.lan.netmask=255.255.252.0
network.lan.gateway=10.20.32.1
network.lan.ip6assign=60
network.@switch[0]=switch
network.@switch[0].name=switch0
network.@switch[0].reset=1
network.@switch[0].enable_vlan=1
network.@switch_vlan[0]=switch_vlan
network.@switch_vlan[0].device=switch0
network.@switch_vlan[0].vlan=1
network.@switch_vlan[0].ports=0 1 2
```

File: nodewatcher/modules/platforms/openwrt/cgm/network.py [hunk 3]
```diff
@@ -243,6 +243,12 @@ def configure_interface(cfg, node, interface, section, iface_name):
             dhcp_rule.dest_port = 68
             dhcp_rule.target = 'ACCEPT'
 
+            # Ensure ICMP traffic is allowed on the uplink interface.
+            icmp_rule = cfg.firewall.add('rule')
```
Comment (mitar) [inline nodewatcher/modules/platforms/openwrt/cgm/network.py]: What are those changes?

Comment (robimarko) [reply nodewatcher/modules/platforms/openwrt/cgm/network.py]: That is one of Kostkos commits.
That should not have been included

**Comment by mitar** [comment]
Also, you should never ever commit into `development` branch in your fork. You should always make a new "feature" branch naming the feature you are adding (like support for UniFi APs) and then work there.

In this way when we add extra stuff to `development`, you will not have problems.

Or when we will not pull all your commits in from this branch. For example. We will probably squash commits when merging this. So then our `development` branch will diverge from yours. And you will have problems.

**Comment by robimarko** [comment]
Yes,no I get it.
I have not been using git a lot so this is pretty much all new now.

**Comment by mitar** [comment]
Yes, maybe go through a tutorial first, and learn it. It is very useful to know more about it.

**Comment by robimarko** [comment]
I think I will close this pull request and make a new one from a proper fork and branch.
I have been trying to somehow fix this but have not been able

**Comment by mitar** [comment]
OK. But learn for the next time.

**Comment by robimarko** [comment]
Sure,will learn for future.
In meanwhile I have created a new PR #38

**Pull request closed without merging**